### PR TITLE
Add puppet-languageserver to servers

### PR DESCRIPTION
**Pull request by Hoogerino**
Hi everybody!

I've added the option to install the puppet-languageserver. Do you have any suggestions or comments, or does it look OK? :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,48 +58,49 @@ end
...
+| html        | html-language-features (pulled directly from the latest VSCode release)                                                        |
+| java        | Eclipse JDTLS with Lombok                                                                                                      |
+| json        | json-language-features (pulled directly from the latest VSCode release)                                                        |
+| kotlin      | kotlin-language-server                                                                                                         |
+| latex       | texlab                                                                                                                         |
+| lua         | (sumneko) lua-language-server                                                                                                  |
+| php         | intelephense                                                                                                                   |
+| puppet      | puppet-languageserver (pulled directly from the latest [GitHub][https://github.com/puppetlabs/puppet-editor-services] release) |
```
Comment (kabouzeid) [inline README.md:85]: please replace `puppet-languageserver (pulled directly from the latest [GitHub][https://github.com/puppetlabs/puppet-editor-services] release)` with just `puppet-editor-services`

File: lua/lspinstall/servers/puppet.lua [hunk 2]
```diff
@@ -0,0 +1,13 @@
+local config = require"lspinstall/util".extract_config("puppet")
+config.default_config.cmd[1] = "./puppet-editor-services/puppet-languageserver"
+
+return vim.tbl_extend('error', config, {
+  install_script = [[
+    name=puppet-editor-services
```
Comment (kabouzeid) [inline lua/lspinstall/servers/puppet.lua:6]: just write `puppet-editor-services` everywhere instead of `$name`

**Review by kabouzeid — CHANGES_REQUESTED** [review]
Looks fine otherwise. Thank you!

File: README.md (hunk 1 above)
Comment (Hoogerino) [reply README.md]: Fixed this in a new commit, take a look!

File: lua/lspinstall/servers/puppet.lua (hunk 2 above)
Comment (Hoogerino) [reply lua/lspinstall/servers/puppet.lua]: Also changed now.

**Comment by dhollinger** [comment]
@kabouzeid can we get an update on this? I've been waiting for this :)

**Comment by kabouzeid** [comment]
I'm getting `LSP[puppet][Warning] An error occured while starting the Language Server. The server has been disabled.`

**Comment by kabouzeid** [comment]
Merging for now because I don't think it's the fault of the installer

**Comment by dhollinger** [comment]
That error usually shows up if the LSP is misconfigured or a "client" doesn't connect to it after X amount of time. 

The LSP installer worked really well for me. Paired with LSPconfig, LunarVim, and a puppet module, it's working great for me:

Peek 2021-08-12 10-44

**Pull request merged**